sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/PartitionPump.cs

```diff
@@ -197,31 +222,49 @@ private async Task StopAsync(PartitionProcessorCloseReason reason)
         ///
         private async Task RunAsync(CancellationToken cancellationToken)
         {
+            Exception unrecoverableException = null;
+
+            // We'll break from the loop upon encountering a non-retriable exception.  The event processor periodically
+            // checks its pumps' status, so it should be aware of when one of them stops working.
+
             while (!cancellationToken.IsCancellationRequested)
             {
-                IEnumerable<EventData> receivedEvents = null;
-
                 try
                 {
-                    receivedEvents = await InnerConsumer.ReceiveAsync(Options.MaximumMessageCount, Options.MaximumReceiveWaitTime, cancellationToken).ConfigureAwait(false);
+                    var receivedEvents = await InnerConsumer.ReceiveAsync(Options.MaximumMessageCount, Options.MaximumReceiveWaitTime, cancellationToken).ConfigureAwait(false);
```

Comment:
nit:  you may want to consider hoisting this outside the `while` loop rather than potentially allocating each tick.   _(the compiler is likely to do this as an optimization, but...)_